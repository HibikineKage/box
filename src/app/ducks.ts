import {Action} from 'redux';
export const REQUEST_SERVER_INFO = 'REQUEST_SERVER_INFO';
export const RECEIVE_SERVER_INFO = 'RECEIVE_SERVER_INFO';
export const SERVRE_INFO_FETCH_SUCCEED = 'SERVER_INFO_FETCH_SUCCEED';
export const SERVER_INFO_FETCH_FAILED = 'SERVER_INFO_FETCH_FAILED';
export interface State {}
const initialState = {};
const reducer = (state : State = initialState, action : Action) => {
  return state;
};
export default reducer;
