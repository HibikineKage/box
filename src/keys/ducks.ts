import { Action } from 'redux';

enum KeyState {}
interface KeyStatus {
  state: KeyState;
}
interface State {
  keyStatuses: Map<string, KeyStatus>;
}
const initialState: State = { keyStatuses: new Map() };
export const reducer = (state: State = initialState, action: Action) => state;
export default reducer;
