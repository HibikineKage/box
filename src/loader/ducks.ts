import { Action } from 'redux';

interface State {}

const initialState: State = {};

export const reducer = (state: State = initialState, action: Action) => {
  return state;
};

