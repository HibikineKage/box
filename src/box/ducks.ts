import { Action } from 'redux';
import { Player } from '../player/ducks';

export const ADD_BOX = 'ADD_BOX';
export interface AddBoxPayload {
  x: number;
  y: number;
  player: Player;
}
export interface AddBoxAction extends Action {
  payload: AddBoxPayload;
}
const initBox = (addBoxPayload: AddBoxPayload): Box => ({
  ...addBoxPayload,
  isFalling: false,
  isAlive: true,
});
export const REMOVE_BOX = 'REMOVE_BOX';
export interface Box {
  player: Player;
  isFalling: boolean;
  isAlive: boolean;
  x: number;
  y: number;
}
export interface State {
  boxes: Box[];
}
const initialState: State = { boxes: [] };
export const reducer = (state: State = initialState, action: Action) => {
  switch (action.type) {
  case ADD_BOX:
    return {
      ...state,
      boxes: [...state.boxes, initBox(action.payload)],
    };
  case REMOVE_BOX:
  default:
    return state;
  }
};
export default reducer;
