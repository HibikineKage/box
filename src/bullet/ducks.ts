import { Action } from 'redux';
import { Player } from '../player/ducks';

export interface Bullet {
  player: Player;
}
export interface State {
  bullets: Bullet[];
}
const initialState = { bullets: [] };
export const reducer = (state: State = initialState, action: Action) => state;
export default reducer;
