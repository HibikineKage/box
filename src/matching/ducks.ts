import { Action } from 'redux';
import { actionCreatorFactory, isType } from 'typescript-fsa';
import { ADD_ROOM_SUCCEED } from '../rooms/ducks';
import { ClientRoom } from '../server/app';

const actionCreator = actionCreatorFactory();
export const MATCH_ROOM_SUCCEED = 'MATCH_ROOM_SUCCEED';
export const matchRoomSucceed = actionCreator<ClientRoom>(MATCH_ROOM_SUCCEED);
export const MATCH_ROOM_FAILED = 'MATCH_ROOM_FAILED';
export const MATCH_ROOM = 'MATCH_ROOM';
export const MATCH_ROOM_TIMEOUT = 'MATCH_ROOM_TIMEOUT';
export const JOIN_ROOM = 'JOIN_ROOM';
export const JOIN_ROOM_SUCCEED = 'JOIN_ROOM_SUCCEED';
export const JOIN_ROOM_TIMEOUT = 'JOIN_ROOM_TIMEOUT';
export const joinRoomSucceed = actionCreator<ClientRoom>(JOIN_ROOM_SUCCEED);
export const JOIN_ROOM_FAILED = 'JOIN_ROOM_FAILED';
export const DELETED_ROOM = 'DELETED_ROOM';
export interface MatchingUser {
  name: string;
}
export interface JoinRoomAction extends Action {
  payload: ClientRoom;
}
export enum MatchingStatus {
  Default = 'DEFAULT',
  Waiting = 'WAITING',
  Joining = 'JOINING',
  Timeout = 'TIMEOUT',
}
export interface State {
  status: MatchingStatus;
}
const initialState: State = {
  status: MatchingStatus.Default,
};
const reducer = (state: State = initialState, action: Action) => {
  if (isType(action, joinRoomSucceed) || isType(action, matchRoomSucceed)) {
    return {
      ...state,
      status: MatchingStatus.Default,
    };
  }
  switch (action.type) {
  case ADD_ROOM_SUCCEED:
    return {
      ...state,
      status: MatchingStatus.Waiting,
    };
  case JOIN_ROOM:
    return {
      ...state,
      status: MatchingStatus.Joining,
    };
  case MATCH_ROOM_FAILED:
  case JOIN_ROOM_FAILED:
    return {
      ...state,
      status: MatchingStatus.Default,
    };
  case MATCH_ROOM_TIMEOUT:
    return {
      ...state,
      status: MatchingStatus.Timeout,
    };
  default:
    return state;
  }
};
export default reducer;
