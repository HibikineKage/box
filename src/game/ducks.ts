import { Action } from 'redux';
import { actionCreatorFactory, isType } from 'typescript-fsa';
import { IPlayer, Player, updatePlayer } from '../player/ducks';
import { Box } from '../box/ducks';
import { Wall } from '../walls/ducks';
import { IBullet } from '../bullet/ducks';
import { matchRoomSucceed, joinRoomSucceed } from '../matching/ducks';
import { ClientRoom } from '../server/app';

const actionCreator = actionCreatorFactory();
export const gameTick = actionCreator('GAME_TICK');
export const START_GAME = 'START_GAME';
export const startGame = actionCreator(START_GAME);
export const COUNT_DOWN = 'COUNT_DOWN';
export const countDown = actionCreator(COUNT_DOWN);
export const INITIAL_COUNT = 5;

export enum GameStatus {
  NotStarting = 'NOT_STARTING',
  InitializeWaiting = 'INITIALIZE_WAITING',
  BeforeWaiting = 'BEFORE_WAITING',
  CountDown = 'COUNT_DOWN',
}

export interface State {
  status: GameStatus;
  countDownNumber: number;
  players: IPlayer[];
}

const initialState = {
  status: GameStatus.NotStarting,
  countDownNumber: 0,
  players: [],
};

const initPlayer = (room: ClientRoom): IPlayer[] => [
  new Player(),
  new Player(),
];

export const reducer = (state: State = initialState, action: Action) => {
  if (isType(action, matchRoomSucceed) || isType(action, joinRoomSucceed)) {
    return {
      ...state,
      players: initPlayer(action.payload),
    };
  }
  if (isType(action, startGame)) {
    return {
      ...state,
      status: GameStatus.InitializeWaiting,
      countDownNumber: INITIAL_COUNT,
    };
  }
  if (isType(countDown, action.type)) {
    return {
      ...state,
      countDownNumber: state.countDownNumber - 1,
    };
  }
  if (isType(action, gameTick)) {
    return {
      ...state,
      players: state.players.map(updatePlayer),
    }
  }
};

export default reducer;
