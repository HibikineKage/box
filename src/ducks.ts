import { combineReducers } from 'redux';
import app, { State as AppState } from './app/ducks';
import router, { State as RouterState } from './router/ducks';
import rooms, { State as RoomsState } from './rooms/ducks';
import matching, { State as MatchingState } from './matching/ducks';
import game, { State as GameState } from './game/ducks';

export interface State {
  app: AppState;
  game: GameState;
  router: RouterState;
  rooms: RoomsState;
  matching: MatchingState;
}

export default combineReducers({ app, router, rooms, matching });
