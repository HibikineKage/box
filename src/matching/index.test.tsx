import * as React from 'react';
import {Props, Matching} from './index';
import * as renderer from 'react-test-renderer';
import {MatchingStatus} from './ducks';
describe('Matching returns ', () => {
  const mockBackToLobby = jest.fn();
  const props : Props = {
    matchingStatus: MatchingStatus.Default,
    backToLobby: mockBackToLobby
  };
  test('matching scene', () => {
    const tree = renderer.create(<Matching {...props}/>);
    expect(tree.toJSON).toMatchSnapshot();
  });
  test('timeout', () => {
    const tree = renderer.create(<Matching {...props} matchingStatus={MatchingStatus.Timeout}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  })
});