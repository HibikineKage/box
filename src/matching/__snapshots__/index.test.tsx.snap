// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matching returns  matching scene 1`] = `[Function]`;

exports[`Matching returns  timeout 1`] = `
<div>
  <div>
    <p>
      Timed out
    </p>
    <p>
      <button
        onClick={[MockFunction]}
      >
        Back to lobby
      </button>
    </p>
  </div>
</div>
`;
